addons/web_editor/static/src/js/wysiwyg/widgets/link_tools.js

```diff
@@ -425,6 +444,18 @@ export class LinkTools extends Link {
     _onClickCheckbox(ev) {
         const $target = $(ev.target);
         $target.closest('we-button.o_we_checkbox_wrapper').toggleClass('active');
+        if ($target[0].getAttribute("name") === "direct_download") {
+            const el = $target.closest("we-button.o_we_checkbox_wrapper")[0];
+            const urlInputEl = this.$el[0].querySelector("#o_link_dialog_url_input");
+            let href = urlInputEl.value;
+            if (el.classList.contains("active")) {
+                href = href.replace("&download=true", "") + "&download=true";
```

Comment:
Why is this necessary?

Reply:
I wanted to be sure that "&download=true" was not appended manually by the user. But I've refactored.

Reply:
Isn't the fact that the state changes automatically changes the state of the checkbox? I just tried manually adding &download=true and the checkbox ticked itself, so I don't think it should be a concern, you can just assume that if the checkbox is ticked, it is present right? and if not, it is not

Anyway I'm okay with the way you did in the refactor if you think it's worth it to avoid buggy links.